src/safeposix/syscalls/sys_calls.rs

```diff
@@ -456,6 +497,9 @@ impl Cage {
             newsigset.insert(0, mainsigset);
         }
 
+        // Initialize a new cage object to replace the current running image
+        // We set all the ids to -1 to indicate the loading phase
```

Comment:
What is the loading phase?

Reply:
it is what the NaCL runtime expects when a cage has just been created - hence anytime a cage is created - the first call to getuid or any similar call results in -1 to ensure commonality of behavior